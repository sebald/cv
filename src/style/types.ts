import { Interpolation, ThemeProps } from 'styled-components';
import { theme } from './theme';

export type Theme = typeof theme;

export type ThemedInterpolation = Interpolation<ThemeProps<Theme>>;
