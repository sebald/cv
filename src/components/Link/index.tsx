export * from './Link';
export * from './IconLink';
