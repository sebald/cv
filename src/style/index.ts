export { default as theme, Theme } from './theme';

export * from './mixins';
export * from './styled-components';
export * from './utils';
