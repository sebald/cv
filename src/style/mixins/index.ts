export * from './animation';
export * from './breakpoints';
export * from './color';
export * from './space';
export * from './typography';
export * from './width';

export * from './types';
