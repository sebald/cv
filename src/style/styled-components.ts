import * as styledComponents from 'styled-components';
import {
  ThemedStyledComponentsModule,
  ThemedStyledFunction,
} from 'styled-components';

import { Theme } from './types';

// https://github.com/styled-components/styled-components/issues/630#issuecomment-317172454
function withProps<U>() {
  return <P, T, O>(
    fn: ThemedStyledFunction<P, T, O>
  ): ThemedStyledFunction<P & U, T, O & U> =>
    fn as ThemedStyledFunction<P & U, T, O & U>;
}

// Alternative, maybe better approach?
type TTag = keyof JSX.IntrinsicElements;

function suit<U>(tag: TTag) {
  return styled[tag] as ThemedStyledFunction<
    JSX.IntrinsicElements[TTag] & U,
    Theme
  >;
}

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<Theme>;

export { styled, css, injectGlobal, keyframes, withProps, suit, ThemeProvider };
