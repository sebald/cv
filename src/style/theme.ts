export const theme = {
  baseline: 12,
  htmlFontSize: 16,
};
