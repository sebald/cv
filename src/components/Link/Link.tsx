import { color, styled } from '../../style';

export const Link = styled.a`
  color: ${({ theme }) => theme.typography.body.color};
  text-decoration: none;
  transition: color 200ms ease-in-out;

  &:hover {
    color: ${color(palette => palette.grey['500'])};
  }
`;
