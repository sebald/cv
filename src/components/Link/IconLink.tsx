import { StyledComponentClass } from 'styled-components';
import { color, Theme } from '../../style';

import { Icon } from '../Icon';
import { Link } from './Link';

const IconWithLink: StyledComponentClass<
  React.ClassAttributes<HTMLAnchorElement> &
    React.AnchorHTMLAttributes<HTMLAnchorElement>,
  Theme
> = Icon.withComponent(Link);

export const IconLink = IconWithLink.extend`
  color: ${color(palette => palette.grey['300'])};

  &:hover {
    color: ${color(palette => palette.grey['600'])};
  }
`;
