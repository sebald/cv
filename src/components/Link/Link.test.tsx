import { shallow } from 'enzyme';
import * as React from 'react';

import { theme } from '../../style';
import { Link } from './Link';

test('render correct', () => {
  const wrapper = shallow(<Link theme={theme} />);
  expect(wrapper).toMatchSnapshot();
});
