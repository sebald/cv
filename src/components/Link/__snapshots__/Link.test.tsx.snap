// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correct 1`] = `
.c0 {
  color: #757575;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 200ms ease-in-out;
  transition: color 200ms ease-in-out;
}

.c0:hover {
  color: #9e9e9e;
}

<a
  className="c0"
/>
`;
